extensions/sql/data/WriteCallback.h

```diff
@@ -0,0 +1,47 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "FlowFileRecord.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+
+class WriteCallback : public OutputStreamCallback {
+public:
+  WriteCallback(const char *data, uint64_t size)
+    : _data(const_cast<char*>(data)),
```

Comment:
If there a reason for taking `const char*` and casting away `const` instead of taking mutable `char*`? If so, please add code comment, otherwise please fix it.

Reply:
WriteCallback called like WriteCallback(out.data(), ...); out is string out.data() type is const char*.
With WriteCallback(char*, ...) , WriteCallback(out.data(), ...) won't compile because cannot convert from const char* -> char*.

But I'll refactor to WriteCallback(const std::string&).

Reply:
In that case I'd prefer to have the `const_cast` near the problematic `.data()` call so that it can be deferred from the context that the reason for casting away const is the missing non-`const` `string::data` overload. Alternatively you can pass `&out[0]` if `out.size() > 0`.

I'm asking for the workaround to be moved closer to the problem it works around so that we can avoid affecting the interface of unrelated code (`WriteCallback` in this case).